src/wp-includes/class-wp-query.php

```diff
@@ -1870,6 +1871,10 @@ public function get_posts() {
 			$q['update_post_term_cache'] = true;
 		}
 
+		if ( ! isset( $q['update_menu_item_cache'] ) ) {
+			$q['update_menu_item_cache'] = false;
```

Comment:
I think the default here should be `true` since this is the existing behavior and goes in line with the rest of the existing flags in this class like `update_post_term_cache` meaning that in order to deactivate it should be explicit, this also prevents a breaking change on existing sites upgrading to this version would continue working as they are with no changes.

```suggestion
			$q['update_menu_item_cache'] = true;
```

Reply:
I am not sure I understand what you mean here when it comes to breaking change. This is a new param. As it is not especially useful for post types outside nav_menu_items, this should be disabled by default.

Reply:
I agree with @spacedmonkey.

Reply:
Great, follow-up question set the value to `true` means is enabled or disabled?

Since I'm a bit confused now, as other values like `suppress_filters` when not set the initial value is `false`